Sources/MapboxMaps/Foundation/DictionaryCoding/DictionaryEncoder.swift

```diff
@@ -0,0 +1,399 @@
+
+    final class KeyedEncodingContainer<Key: CodingKey>: KeyedEncodingContainerProtocol, EncoderContainer {
+        let codingPath: [CodingKey]
+        private let userInfo: [CodingUserInfoKey: Any]
+        private var storage: [String: Storage] = [:]
+
+        init(codingPath: [CodingKey], userInfo: [CodingUserInfoKey: Any]) {
+            self.codingPath = codingPath
+            self.userInfo = userInfo
+        }
+
+        func toAny() throws -> Any {
+            try storage.mapValues { try $0.toAny() }
+        }
+
+        func encodeNil(forKey key: Key) throws {
+            storage[key.stringValue] = .value(Any?.none as Any)
+        }
+
+        func encode(_ value: Bool, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Int, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Int8, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Int16, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Int32, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Int64, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: UInt, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: UInt8, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: UInt16, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: UInt32, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: UInt64, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: String, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Float, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode(_ value: Double, forKey key: Key) throws {
+            storage[key.stringValue] = .value(value)
+        }
+
+        func encode<T: Encodable>(_ value: T, forKey key: Key) throws {
+            if DictionaryEncoder.isSupportedType(T.self) {
+                storage[key.stringValue] = .value(value)
+            } else {
+                let result = try Encoder(codingPath: codingPath.appending(key: key), userInfo: userInfo).encode(value)
+                storage[key.stringValue] = .value(result)
+            }
+        }
+
+        func encodeIfPresent<T: Encodable>(_ value: T?, forKey key: Key) throws {
```

Comment:
It seems that something is wrong with this - this method does not seem to be called for primitive types(like Double - e.g. minzoom is not encoded).

Reply:
my bad, I didn't commit `encodeIfPresent` for primitive types :-D